interface/ceed-vector.c

```diff
@@ -1070,6 +1070,39 @@ int CeedVectorRestoreArrayReadFP64(CeedVector vec, const double **array) {
   return CeedVectorRestoreArrayReadGeneric(vec, (const void **) array);
 }
 
+/**
+  @brief Check the current status of the CeedVector's data arrays.
+
+  This function sets unsigned int parameters indicating for which precisions
+  the data for mem_type is not NULL, for borrowed and owned arrays, and also
+  which precisions are currently valid.
+  The values can be checked bitwise, such that, e.g.,
+  (valid_status & (1 << CEED_SCALAR_FP32)) will be true if the FP32 array is
+  currently valid, and false otherwise.
+
+  @param vec              CeedVector for which to check status
+  @param mem_type         Mem type for which to check status
+  @param valid_status     Status indicator for valid data
+  @param borrowed_status  Status indicator for borrowed arrays
+  @param owned_status     Status indicator for owned arrays
+**/
+int CeedVectorCheckArrayStatus(CeedVector vec, CeedMemType mem_type,
```

Comment:
I added this function to the interface level, thinking it would be used by tests for multiprecision vector functionality (and potentially, users may want this feature).  
It also has a backend implementation, so it's added to the `CeedVector_private` struct.  Since this affects the user interface, I wanted to make sure and point it out to see what people thought.

Reply:
If we keep this, we may also want an interface function for convenient "pretty" printing of status for all vector arrays, or we could implement such a function in the tests directly.